stout/flags.cc

```diff
@@ -99,8 +205,69 @@ void Parser::Parse(int* argc, const char*** argv) {
       break;
     }
 
-    // Skip anything that doesn't look like a flag.
+    // Skip anything that doesn't look like a flag, subcommand or positional
+    // argument.
     if (arg.find("--") != 0) {
+      // It could be probably subcommand.
+      google::protobuf::Message* subcommand = TryParseSubcommand(arg);
+      if (subcommand != nullptr) {
+        Parser::subcommands.emplace(arg, cur_subcommand_message_);
+        std::unique_ptr<Parser> nested_parser =
+            std::make_unique<Parser>(Parser::Builder(subcommand).Build());
```

Comment:
We want any build errors from creating the nested parser to happen when the outtermost parser `Build()` is called not now at runtime when we're calling `Parse()`.

Also, what about the parser options that were added to the outermost parser, we'll need to add them here too!